### DISCUSSION TITLE
Loading embedding registry for inference?

### DISCUSSION THREAD

**Discussion by akashsara** [lancedb/lancedb#1732] [Q&A]
Hello! I have a lancedb table created, an embedding registry setup and a bunch of data ingested into it. Now when I want to query the database, it seems like the embedding model is not loaded when I open the table, it happens the first time I send a query. Is there a way to explicitly load the model beforehand? Otherwise the first query always takes a lot longer as the embedding model is being loaded in the background.

Right now I'm just running a dummy query after opening the table to bypass this but I was wondering if there's a better solution?

**Comment by TayfurYldz** [comment]
Opening the table restores the **embedding-function configuration**, but it does not necessarily materialize the underlying ML model. Those are separate steps.

On current LanceDB, `table.embedding_functions` is reconstructed from the table's schema metadata. A text vector query then reaches `conf.function.compute_query_embeddings_with_retry(...)`. For the built-in `sentence-transformers` function specifically, the actual `SentenceTransformer(...)` object is lazy:

```python
@property
def embedding_model(self):
    return self.get_embedding_model()

@weak_lru(maxsize=1)
def get_embedding_model(self):
    ...
    return sentence_transformers.SentenceTransformer(...)
```

So the first operation that needs the model pays the load cost, and subsequent operations in that process reuse the cached model. That matches the first-query latency you are seeing.

There is not a generic `EmbeddingFunction.load()`/preload hook in the base embedding-function interface that LanceDB calls when a table is opened. If you know you are using the sentence-transformers implementation, you can explicitly warm that implementation without executing a database query:

```python
table = db.open_table("documents")

for conf in table.embedding_functions.values():
    func = conf.function
    if hasattr(func, "embedding_model"):
        _ = func.embedding_model
```

I would keep that implementation-specific rather than assuming every embedding provider has a local model to load; API-backed embedding functions have different startup behavior.

If you want provider-agnostic application code, your existing warm-up embedding/query is therefore a reasonable boundary: force one embedding computation during application startup/readiness instead of letting the first user request pay for it. The important point is that reopening the table restores the registered function from metadata; it does not promise eager initialization of that function's external/model resources.
